src/rust/ensogl/src/display/style/sheets.rs

```diff
@@ -0,0 +1,1063 @@
+//! This module defines a cascading style sheet registry and related style management utilities.
+
+use crate::prelude::*;
+
+use crate::control::callback;
+use crate::data::HashMapTree;
+use crate::data::Index;
+use crate::data::OptVec;
+
+pub use super::data::Data;
+pub use super::data::data;
+pub use super::path::Path;
+
+
+
+// =============
+// === Query ===
+// =============
+
+/// Pointer to a sheet. Always bound to the most specific style sheet matching the query (if such
+/// exists). For example, the query 'panel.button.size' will be bound to 'panel.button.size',
+/// 'button.size', or 'size' in that order.
+///
+/// # Implementation Details
+/// Each query keeps a list of all style sheets which match this query (`matches`). For example,
+/// the query 'panel.button.size' contains three matches - 'panel.button.size', 'button.size', and
+/// 'size'. The longest match with a defined value is considered the best match and is remembered
+/// (`binding`). Moreover, each query keeps list of all sheets which use this query in their
+/// expressions (`usages`). A query is considered unused and can be safely removed from the graph
+/// if no sheets use it in their expressions and it is not referred by an external, user code
+/// (`external_count`).
+#[derive(Debug)]
+pub struct Query {
```

Comment:
So, this was before `Var`? I really like this change :+1:

Reply:
yes